m3ntorship/charity-nextjs
# Feature/introduce error case no content

### Bug description

- If one API request is failed,other requested are not processed and server responds with error
- if the website content is empty the server resonds with error

### solution approach
- handling error in each API request uses a promise chain.
- return `statusCode` value in case the response is empty or due to network failure or bad request.
- checking at the component level if the data contains`stausCode` field then.

```
       getCharityAPI(" end point "),.then(res => {
          //return response in case response is not empty
          if (Object.keys(res.data).length) {
            return res;
          } else {
            return {
              //in case reoonse is {} or []
              data: {
                statusCode: 404
              }
            };
          }
        })
         // in case other errors
        .catch(err => ({
          data: {
            statusCode: 404
          }
        }))
```
### How to test
- delete content of some sections Ex: `main-carousels`
- type wrong API end point Ex `/main-caro`
          

> - The section should not appear ( we can display dynamic content based on the error)

### Still in progress
- `articles,dev-team,causes`
